Sources/StreamVideo/WebRTC/WebRTCClient.swift

```diff
@@ -595,24 +698,33 @@ class WebRTCClient: NSObject {
         webSocketClient.onWSConnectionEstablished = { [weak self] in
             guard let self = self else { return }
             Task {
-                try await self.handleSocketConnected()
+                if isMigrating {
+                    await self.sendMigrationJoinRequest()
+                } else {
+                    try await self.handleSocketConnected()
+                }
             }
         }
 
         return webSocketClient
     }
     
     private func handleSocketConnected() async throws {
+        let sdp = try await tempOfferSdp()
+        await sendJoinRequest(with: sdp)
+    }
+    
+    private func tempOfferSdp() async throws -> String {
```

Comment:
nit: Can we find a more descriptive name for this one?

Reply:
we can't really, since it's a temporary offer sdp 😅 It is a temp sdp that we exchange the first time with an SFU, and then it's thrown away. Let me know if you have a better idea for the name.

Reply:
If that's as descriptive as it goes I'm good with it